### DISCUSSION TITLE
Dynamically replace a part of URL

### DISCUSSION THREAD

**Discussion by alexbn71** [DaniloNovakovic/chrome-dynamic-bookmarks#37] [Ideas]
I'd like to replace a part of URL with the result of a GET query in this way:

**http://[myip]/folder/page.html**

and [myip] placeholder raplaced by the result of the query ie... 

**GET http://findmyip.site?token=hdjhd&param1=x&param=y**

that return a string like "186.73.22.11", so finally the bookmark point to 

**http://186.73.22.11/folder/page.html**

this can be useful to avoid firewall restriction inside companies that now often block all dynamic dns services like no-ip.com, dydns.com etc so a url like **http://myhome.ddns.net/folder/page.html** is no longer accessible.

Thanks

------------------------------------------------------------------------------------------------
_Issue resolved with Chrome... using javascript code inside the URL field of a bookmark._

**Comment by DaniloNovakovic** [comment]
Hey @alexbn71. I am currently working on version 3.0.0 which is a complete rework of this extension with the goal of making it more accessible and customizable. I've realized that some people might prefer something other than regular expressions which for future versions I plan on making easily configurable.

Current plans are to support the unix-like [wildcard patterns](https://ryanstutorials.net/linuxtutorial/wildcards.php) with the possibility to create, export and import additional custom patterns.

My initial idea is to have a JSON object where the key would be mapped to a regular expression or a string (this should be either loadable from the file or modifiable in some kind of options page).

For example:

```javascript
{
	"$numbers: "\d+"
}
```
would allow you to write something like `https://mysite.com/$numbers` which would internally get mapped to `https:\/\/mysite\.com\/\d+` regular expression.

This would allow you to also use:

```javascript
{
	"[myid]": "186.73.22.11"
}
```

The feature mentioned above was going to be specifically aimed for use in the search pattern that would define if the URL should be updated or not, but I think it could be easily extendable to possibly replace URL as well.

Something like the following could work:

```javascript
{
	"[myid]": {
		"pattern": "186.73.22.11",
		"replaceUrl": true  /*or string to replace with (ex. "localhost")*/
	}
}
```

>> This is just an example of a proposal on how the future JSON file could look like. If you have a wish/suggestion for the syntax feel free to post it below.

As of now my priority is getting version 3.0.0 working and running so, for now, I will have to put your request on hold, but know that I will definitely come back to it at some point (maybe for version 3.1? 🤔).
